File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        maxprofit = 0
        min = prices[0]
        for i in range(len(prices)):
            if prices[i] < min:
                min = prices[i]
            if prices[i] - min >  maxprofit:
                maxprofit = prices[i] - min

        return maxprofit          

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function (prices) {
    let min = prices[0], diff = 0;
    for (let i in prices) {
        min = Math.min(min, prices[i]);
        diff = Math.max(diff, (prices[i] - min));
    }
    return diff
};